app/models/comment.php

```diff
@@ -0,0 +1,48 @@
+<?php
+    class Comment extends AppModel
+    {
+        public $validation = array(
+
+            'username' => array(
+                'length' => array(
+                    'validate_between', MIN_USERNAME_CHARACTERS, MAX_USERNAME_CHARACTERS,
+                ),
+            ),
+
+            'body' => array(
+                'length' => array(
+                    'validate_between', MIN_BODY_CHARACTERS, MAX_BODY_CHARACTERS,
+                ),
+            ),
+        );
+
+        // Fetches all comments from Database
+        public static function getAll ($thread_id)
+        {
+            $db = DB::conn();
+            $rows = $db->rows(
+                'SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC', array($thread_id));
+
+            foreach ($rows as $row){
+                $comments[] = new Comment($row);
```

Comment:
please declare $comments initially as empty array to avoid errors in case $rows is empty